sql/scan.go

```diff
@@ -231,22 +235,39 @@ func (n *scanNode) initScan() bool {
 		return true
 	}
 
-	// Retrieve all of the keys that start with our index key prefix.
-	if len(n.startKey) == 0 {
-		n.startKey = proto.Key(MakeIndexKeyPrefix(n.desc.ID, n.index.ID))
-	}
-	if len(n.endKey) == 0 {
-		n.endKey = n.startKey.PrefixEnd()
+	if len(n.spans) == 0 {
+		// If no spans were specified retrieve all of the keys that start with our
+		// index key prefix.
+		start := proto.Key(MakeIndexKeyPrefix(n.desc.ID, n.index.ID))
+		n.spans = append(n.spans, span{
+			start: start,
+			end:   start.PrefixEnd(),
+		})
 	}
+
+	// Retrieve all the spans.
+	b := &client.Batch{}
```

Comment:
Ok, I'm aware of that. I thought the `TODO` above this method already captured this. I seem to have written that TODO multiple times so didn't think it was necessary calling out `LIMIT` here. Note that the limiting the rows read is only possible if the scan order matches the requested order.